pkg/window/strategy/fixed/fixed.go

```diff
@@ -77,74 +77,50 @@ func (f *Fixed) AssignWindow(eventTime time.Time) []window.AlignedKeyedWindower
 	}
 }
 
-// CreateWindow adds a window for a given interval window
-func (f *Fixed) CreateWindow(kw window.AlignedKeyedWindower) window.AlignedKeyedWindower {
+// InsertIfNotPresent inserts a window to the list of active windows if not present and returns the window
+func (f *Fixed) InsertIfNotPresent(kw window.AlignedKeyedWindower) (aw window.AlignedKeyedWindower, isPresent bool) {
 	f.lock.Lock()
 	defer f.lock.Unlock()
 
 	// this could be the first window
 	if f.entries.Len() == 0 {
 		f.entries.PushFront(kw)
-		return kw
+		return kw, false
 	}
 
 	earliestWindow := f.entries.Front().Value.(*keyed.AlignedKeyedWindow)
 	recentWindow := f.entries.Back().Value.(*keyed.AlignedKeyedWindow)
 
-	// late arrival
-	if !earliestWindow.StartTime().Before(kw.EndTime()) {
+	// if there are only upto 2 windows in the list
+	if recentWindow.StartTime().Equal(kw.StartTime()) && recentWindow.EndTime().Equal(kw.EndTime()) {
+		aw = recentWindow
+		isPresent = true
+	} else if !earliestWindow.StartTime().Before(kw.EndTime()) {
+		// late arrival
 		f.entries.PushFront(kw)
+		aw = kw
 	} else if !recentWindow.EndTime().After(kw.StartTime()) {
 		// early arrival
 		f.entries.PushBack(kw)
+		aw = kw
 	} else {
 		// a window in the middle
-		for e := f.entries.Back(); e != nil; e = e.Prev() {
+		for e := f.entries.Front(); e.Next() != nil; e = e.Next() {
```

Comment:
Why are we reversing the order?

Reply:
we don't have to reverse the order, will change it

Reply:
done